src/Microsoft.ML.StandardLearners/Standard/MultiClass/MetaMulticlassTrainer.cs

```diff
@@ -2,56 +2,97 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using Float = System.Single;
-
+using System;
+using System.Linq;
 using Microsoft.ML.Runtime.CommandLine;
 using Microsoft.ML.Runtime.Data;
 using Microsoft.ML.Runtime.Data.Conversion;
 using Microsoft.ML.Runtime.EntryPoints;
 using Microsoft.ML.Runtime.Internal.Calibration;
 using Microsoft.ML.Runtime.Internal.Internallearn;
 using Microsoft.ML.Runtime.Training;
+using Microsoft.ML.Core.Data;
 
 namespace Microsoft.ML.Runtime.Learners
 {
-    using TScalarTrainer = ITrainer<IPredictorProducing<Float>>;
+    using TScalarTrainer = ITrainerEstimator<IPredictionTransformer<IPredictorProducing<float>>, IPredictorProducing<float>>;
```

Comment:
>IPredictionTransformer [](start = 45, length = 22)

should be BinaryPredictionTransformer

Reply:
No, it's fine as it is

---
In reply to: [217218242](https://github.com/dotnet/machinelearning/pull/865#discussion_r217218242) [](ancestors = 217218242)

Reply:
Binary is not an interface, so we can't have covariance if we have it there

---
In reply to: [217234106](https://github.com/dotnet/machinelearning/pull/865#discussion_r217234106) [](ancestors = 217234106,217218242)